src/controls/qml/RemorseTimer.qml

```diff
@@ -0,0 +1,265 @@
+        id: countdownArc
+        anchors {
+            horizontalCenter: parent.horizontalCenter
+            verticalCenter: parent.verticalCenter
+        }
+        width: Dims.l(22)
+        height: width
+        segmentAmount: remorseTimer.gaugeSegmentAmount
+        inputValue: remorseTimer.arcValue
+        fgColor: "#ffffff"
+        bgColor: Qt.rgba(1, 1, 1, 0.2)
+        start: remorseTimer.gaugeStartDegree
+        endFromStart: remorseTimer.gaugeEndFromStartDegree
+    }
+
+    Label {
+        id: countdownLabel
+        anchors.centerIn: countdownArc
+        font {
+            pixelSize: Dims.l(18)
+            styleName: "SemiBoldCondensed"
+        }
+        color: "#ffffff"
+        text: remorseTimer.countdownSeconds // Direct integer display
+        z: countdownArc.z + 1
+    }
+
+    Label {
+        id: actionLabel
+        anchors {
+            horizontalCenter: parent.horizontalCenter
+            bottom: countdownArc.top
+            bottomMargin: Dims.l(1)
+        }
+        font.pixelSize: Dims.l(6)
+        color: "#ffffff"
+        text: action
+    }
+
+    Label {
+        id: cancelLabel
+        anchors {
+            horizontalCenter: parent.horizontalCenter
+            top: countdownArc.bottom
+            topMargin: Dims.l(1)
+        }
+        font.pixelSize: Dims.l(6)
+        color: "#ffffff"
+        text: cancelText
+    }
+
+    MouseArea {
+        anchors.fill: parent
+        onClicked: {
+            timer.stop();
+            fadeAnimation.stop();
+            arcAnimation.stop();
+            syncTimer.stop();
+            fadeOutAnimation.start();
+        }
+    }
+
+    NumberAnimation {
+        id: fadeAnimation
+        target: remorseTimer
+        property: "opacity"
+        from: 0
+        to: 1
+        duration: 300
+    }
+
+    NumberAnimation {
+        id: fadeOutAnimation
+        target: remorseTimer
+        property: "opacity"
+        from: 1
+        to: 0
+        duration: 300
+        easing.type: Easing.InOutQuad
+        onStopped: {
+            remorseTimer.visible = false;
+            remorseTimer.cancelled();
+        }
+    }
+
+    NumberAnimation {
+        id: arcAnimation
+        target: remorseTimer
+        property: "arcValue"
+        from: 100
+        to: 0
+        duration: remorseTimer.interval
+        easing.type: Easing.Linear
+    }
+
+    Timer {
+        id: syncTimer
+        interval: remorseTimer.interval / Math.floor(remorseTimer.interval / 1000) // 1000ms for 3000ms
+        running: false
+        repeat: true
+        property int steps: Math.floor(remorseTimer.interval / 1000) // 3 steps
+        onTriggered: {
+            steps--;
+            remorseTimer.countdownSeconds = steps;
+            if (steps <= 0) {
+                syncTimer.stop();
+            }
+        }
+    }
+
+    // Start the timer and show the overlay
+    function start() {
+        countdownSeconds = Math.floor(interval / 1000); // Reset to 3
```

Comment:
What if the user doesn't use the `interval = 3`. Does it still reset to 3? :wink: